tests/integrationv2/test_hello_retry_requests.py

```diff
@@ -0,0 +1,76 @@
+import copy
+import os
+import pytest
+import time
+
+from configuration import available_ports, TLS13_CIPHERS, ALL_TEST_CURVES, ALL_TEST_CERTS, HRR_CLIENT_KEYSHARES
+from common import ProviderOptions, Protocols, data_bytes, Curves
+from fixtures import managed_process
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name
+   
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("keyshare", HRR_CLIENT_KEYSHARES, ids=get_parameter_name)
+def test_hrr_with_s2n_as_client(managed_process, cipher, provider, curve, protocol, certificate, keyshare):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(64)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        data_to_send=random_bytes,
+        insecure=True,
+        curve=curve,
+        extra_flags=keyshare,
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.extra_flags = None
+    server_options.curve = Curves.X25519
+
+    # Passing the type of client and server as a parameter will
+    # allow us to use a fixture to enumerate all possibilities.
+    server = managed_process(provider, server_options, timeout=5)
+    client = managed_process(S2N, client_options, timeout=5)
+
+    # The client should connect and return without error
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert bytes("Curve: {}".format("x25519").encode('utf-8')) in results.stdout
+
+    marker_found = False
+    supported_groups_found = False
+    shared_group_found = False  
+    data_to_send_found = False 
+    marker_part1 = b"cf 21 ad 74 e5"
+    marker_part2 = b"9a 61 11 be 1d"
+
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+
+        if marker_part1 in results.stdout and marker_part2 in results.stdout:
+           marker_found = True
```

Comment:
Why not make this an assert? Same for the others, if they're always required to be true.

Reply:
opps, missed that in the clean-up, will change.